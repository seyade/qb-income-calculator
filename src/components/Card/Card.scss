.ui-card {
  position: relative;
  width: 600px;
  padding: 24px 44px;
  font-family: 'Open sans';
  background-color: rgba(236, 238, 241, 1);

  &__title {
    font-size: 28px;
    font-weight: 800;
  }
}
