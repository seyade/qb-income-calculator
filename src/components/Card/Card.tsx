import React, { ReactNode, ReactElement } from 'react';
import './Card.scss';

export interface CardProps {
  title?: string;
  children: ReactNode;
}

function Card({ title, children }: CardProps): ReactElement {
  return (
    <div className="ui-card">
      <h3 className="ui-card__title">{title}</h3>
      <section className="ui-card__body">{children}</section>
    </div>
  );
}

export default Card;
