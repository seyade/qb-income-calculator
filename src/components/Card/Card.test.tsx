import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import Card from './Card';

configure({ adapter: new Adapter() });

describe('Card', () => {
  const wrapper = shallow(
    <Card title="My card">
      <p>Card content</p>
    </Card>
  );

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('has card title', () => {
    expect(wrapper.find('h3').text()).toEqual('My card');
  });

  it('has card content', () => {
    expect(wrapper.find('section').children().text()).toContain('Card content');
  });
});
