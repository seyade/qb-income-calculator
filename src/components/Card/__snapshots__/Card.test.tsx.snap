// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  className="ui-card"
>
  <h3
    className="ui-card__title"
  >
    My card
  </h3>
  <section
    className="ui-card__body"
  >
    <p>
      Card content
    </p>
  </section>
</div>
`;
